src/librustc_codegen_ssa/traits/builder.rs

```diff
@@ -90,6 +90,12 @@ pub trait BuilderMethods<'a, 'tcx: 'a>:
     fn shl(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
     fn lshr(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
     fn ashr(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nswadd(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nuwadd(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nswsub(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nuwsub(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nswmul(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
+    fn nuwmul(&mut self, lhs: Self::Value, rhs: Self::Value) -> Self::Value;
```

Comment:
If we're putting those methods here, we should be using more Rust-y names.
Maybe `unchecked_{u,s}{add,sub,mul}`? Although I feel like "unchecked" might not be conveying that well that "overflow is UB".

Reply:
While I do agree, I am unable to think of another succinct name.